float64 rho_start
float64 rho_end
uint16 npt
uint32 max_f_evals
